pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/column/PhysicalColumnIndexContainer.java

```diff
@@ -133,7 +133,12 @@ public PhysicalColumnIndexContainer(SegmentDirectory.Reader segmentReader, Colum
       _rangeIndex = null;
     }
 
-    PinotDataBuffer fwdIndexBuffer = segmentReader.getIndexFor(columnName, ColumnIndexType.FORWARD_INDEX);
+    // Setting the 'fwdIndexBuffer' to null if forward index is enabled. No-op index readers will be setup for the
+    // forward index disabled columns which doesn't require the 'fwdIndexBuffer'.
+    boolean forwardIndexDisabled = !segmentReader.hasIndexFor(columnName, ColumnIndexType.FORWARD_INDEX);
+    PinotDataBuffer fwdIndexBuffer = forwardIndexDisabled ? null
+        : segmentReader.getIndexFor(columnName, ColumnIndexType.FORWARD_INDEX);
```

Comment:
Can we replace 2 calls with just 1 call to `getIndexFor` ?

Reply:
I can only do this if I put the `getIndexFor` in a `try catch` block. The underlying implementation of the `ColumnIndexDirectory` throws an exception if the index is not present in the `getIndexFor` call. Do you want me to add this in a `try catch` block like this?:

```
PinotDataBuffer fwdIndexBuffer;
try {
    fwdIndexBuffer = segmentReader.getIndexFor(columnName, ColumnIndexType.FORWARD_INDEX);
} catch (Exception e) {
    fwdIndexBuffer = null;
}
```

e.g. of the type of exception thrown down the line (`SingleFileIndexDirectory`):

```
  private PinotDataBuffer checkAndGetIndexBuffer(String column, ColumnIndexType type) {
    IndexKey key = new IndexKey(column, type);
    IndexEntry entry = _columnEntries.get(key);
    if (entry == null || entry._buffer == null) {
      throw new RuntimeException(
          "Could not find index for column: " + column + ", type: " + type + ", segment: " + _segmentDirectory
              .toString());
    }
    return entry._buffer;
  }
```

Reply:
I see. No need for this I guess. Leave it as is